swarm/src/handler/select.rs

```diff
@@ -98,6 +99,80 @@ impl<TProto1, TProto2> ConnectionHandlerSelect<TProto1, TProto2> {
     }
 }
 
+impl<S1OOI, S2OOI, S1OP, S2OP>
+    DialUpgradeError<
+        EitherOutput<S1OOI, S2OOI>,
+        EitherUpgrade<SendWrapper<S1OP>, SendWrapper<S2OP>>,
+    >
+where
+    S1OP: OutboundUpgradeSend,
+    S2OP: OutboundUpgradeSend,
+    S1OOI: Send + 'static,
+    S2OOI: Send + 'static,
+{
+    fn transpose(self) -> Either<DialUpgradeError<S1OOI, S1OP>, DialUpgradeError<S2OOI, S2OP>> {
```

Comment:
I like that you split this out. I actually think it makes sense to split this out for all structs of `ConnectionEvent`. Can we reuse this as part of implementing the bigger `transpose` function?

Reply:
Ah, thanks :D but I did it because in some variants like `ListenUpgradeError` or `AddressChange` transpose doesn't win us anything as we have to call it for both `Select` variants. I updated the code implementing for the variants where it makes sense, ptal Thomas and see if it makes sense to you.